chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,3404 @@
+                    rs = RadioSetting(entry[0], entry[1], rsvl)
+                elif setting['type'] == 'bool':
+                    rs = RadioSetting(entry[0], entry[1],
+                                      RadioSettingValueBoolean(False))
+                elif setting['type'] == 'string':
+                    params = {}
+                    if 'charset' in setting:
+                        params['charset'] = setting['charset']
+                    else:
+                        params['charset'] = chirp_common.CHARSET_ASCII
+                    minlen = 'min_len' in setting and setting['min_len'] or 0
+                    dummy = params['charset'][0] * minlen
+                    rsvs = RadioSettingValueString(minlen,
+                                                   setting['max_len'],
+                                                   dummy,
+                                                   False, **params)
+                    rs = RadioSetting(entry[0], entry[1], rsvs)
+                elif setting['type'] == 'integer':
+                    rs = RadioSetting(entry[0], entry[1],
+                                      RadioSettingValueInteger(setting['min'],
+                                                               setting['max'],
+                                                               setting['min']))
+                elif setting['type'] == 'map':
+                    rsvs = RadioSettingValueMap(setting['map'],
+                                                setting['map'][0][1])
+                    rs = RadioSetting(entry[0], entry[1], rsvs)
+                else:
+                    rs = setting['type'](entry[0], entry[1], self)
+                rs.kenwood_d7_loaded = False
+                grp.append(rs)
+                self._setcache[entry[0]] = rs
+        return grp
+
+    def _do_prerequisite(self, cmd, get, do):
+        if get:
+            arg = 'get_requires'
+        else:
+            arg = 'set_requires'
+        setting = self._get_setting_data(cmd)
+        if arg not in setting:
+            return
+        # Undo prerequisites in the reverse order they were applied
+        if not do:
+            it = reversed(setting[arg])
+        else:
+            it = iter(setting[arg])
+        for prereq, value in it:
+            entry = self._setcache[prereq]
+            if not entry.kenwood_d7_loaded:
+                self._refresh_setting(entry)
+            if do:
+                entry.kenwood_d7_old_value = entry.value.get_value()
+                entry.value.set_value(value)
+            else:
+                entry.value.set_value(entry.kenwood_d7_old_value)
+            sdata = self._get_setting_data(prereq)
+            self._set_setting(prereq, sdata, entry)
+
+    def _drain_input(self):
+        oldto = self.pipe.timeout
+        self.pipe.timeout = 0
+        # Read until a timeout
+        while len(self.pipe.read(1)):
+            pass
+        self.pipe.timeout = oldto
```

Comment:
So, FYI, on Windows changing the timeout apparently clears the input buffer. I can test this on windows too to be sure, but just be super careful about when you do this when there might be data pending you want to read.

Reply:
That's fine here, since clearing the input buffer is the intent, but thanks for the heads-up.